crates/sui-indexer/src/handlers/objects_snapshot_processor.rs

```diff
@@ -221,26 +212,43 @@ impl ObjectsSnapshotProcessor {
                         .await?
                         .unwrap_or_default();
 
+                    // Fetch new changes if buffer is empty
+                    if buffered_changes.is_empty() {
+                        if let Some(new_changes) = stream.next().await {
+                            buffered_changes.extend(new_changes);
+                            // Can't really guarantee order, so we sort checkpoints ascending here
+                            buffered_changes.sort_by_key(|change| change.checkpoint_sequence_number);
```

Comment:
indeed here it's out of order, but to really fix it, we will need to handle a case, when `start_cp + 1` arrives really late, for example start_cp is 100, we will want to wait for 101, before persisting any later checkpoints, there are similar codes around  https://github.com/MystenLabs/sui/blob/main/crates/sui-indexer/src/handlers/committer.rs#L57-L70